### DISCUSSION TITLE
Run AutoConnect as an external file from an existing .ino file

### DISCUSSION THREAD

**Discussion by haldi4803** [Hieromon/AutoConnect#475] [Q&A] [answered]
Hi,

i thought it was a good idea to seperate the AutoConnect part from my main Script as it only starts up when i cannot connect to WiFi.
Tried to create a .h file and renamed the simple example to .cpp

Changed the Setup to void setupHotspot() and the loop to void loopserver() with a while loop cuz it's not the main loop() function that will get looped....
`void loopserver() {
  while (1) {
  Portal.handleClient();
  }
}`

Then made the header file 
```cpp
//
// AutoConnectHotspot.h
//

//Full Function
void setupHotspot();

void loopserver();`

and called both functions
`  Serial.println("WiFi Failed Script");
  if (rtc_get_reset_reason(0) == DEEPSLEEP_RESET) {
      Serial.println("Rewake after Button Press");
      setupHotspot();
      loopserver();
}
```

But.... the Serial Monitor only shows squares output.
![image](https://user-images.githubusercontent.com/17104473/156920242-4af3b6ce-67ee-406b-9cc3-4b169b20b291.png)

I'm not really a pro at programing, did i take the wrong aproach? or miss something obvious?

[OWM_EPD47_epaper_v2.9.ino.txt](https://github.com/Hieromon/AutoConnect/files/8192404/OWM_EPD47_epaper_v2.9.ino.txt)
[AutoConnectHotspot.cpp.txt](https://github.com/Hieromon/AutoConnect/files/8192405/AutoConnectHotspot.cpp.txt)
[AutoConnectHotspot.h.txt](https://github.com/Hieromon/AutoConnect/files/8192406/AutoConnectHotspot.h.txt)

**Comment by Hieromon** [comment, marked as answer]
This is a general programming question, and not an issue related to AutoConnect's library features in the first place.
It will be forwarded to [Discussions](https://github.com/Hieromon/AutoConnect/discussions) and hope for advice from someone.

**Comment by Hieromon** [comment]
@haldi4803 If you post the actual source code just for your convenience, you have very little chance of getting advice. Your post does not take the necessary steps to get advice from others. Tracing code where other libraries are included just for your convenience is not what anyone else likes to do.
To be able to get more advice, I suggest you read the [guide](https://hieromon.github.io/AutoConnect/faq.html#4-reports-the-issue-to-autoconnect-github-repository) in the FAQ to reduce the problem to the smallest code that can be reproduced. To do this, I suggest you first look for the source of the problem by eliminating individual libraries.

FYI, your code is calling `Serial.begin(115200)` twice in the process through `InitialiseSystem()` to `setupHotspot()`. If it is not intentional, then its processing sequence is inappropriate. I suggest you do a careful code trace by yourself first.

**Comment by haldi4803** [comment]
The Problem i went this approach was because i failed to properly integrade the AutoConnect into my script. Somehow threw all other libraries over.
Then my stupid self just thought "i have a working AutoConnect .ino and a Working Weather.ino why not just start one from another?" But well yeah... that doesn't seem to work so easily in Arduino....
After trying with the library approach for a while i gave it up as that seems to be over my head.

Updated to 1.3.4 and took an example from the Hello script and cut it down to be simpler... (seriously, even the simple example is overloaded, that should be renamed for Timezone or such.)

Integrated it into my Script and now it works! Thanks.
[AutoConnect_Hello_custom.ino.txt](https://github.com/Hieromon/AutoConnect/files/8192745/AutoConnect_Hello_custom.ino.txt)


Now that i got that working it's time to get persistence working.
Coming from this discussion: https://github.com/Hieromon/AutoConnect/issues/183

Where would be the best Point to save the vlaues? The ACSubmit Apply button would be my first guess.
But would i have to change the underlying function for that?
the `portal.on(AUX_SAVE_URI, saveParams);` doesnt look right as i use `ACSubmit(Apply, "Apply", SETTINGS_URI);` but then again opening the settings page also uses the same.
